test/mm/buddy.c

```diff
@@ -10,11 +10,9 @@
  */
 #include <test.h>
 
-#include <framework/rng.h>
-
 #include <lp/lp.h>
 #include <mm/buddy/buddy.h>
-#include <mm/model_allocator.h>
+#include "../mock.h"
```

Comment:
For this to work, I had to include the `test` path with `target_include_directories(test_mm PRIVATE .)`, which I don't like because it bypasses the encapsulation of the test framework. I think it'd be better to handle it, by including a function to extend the include path in the framework. Agreed?

Reply:
I mean: rather than showing something like `test_program_include_directories`, I would have some `test_include_path` that may extent the include path for all tests. The rationale for this is that you are "extending" the testing framework with some capabilities. This may work well for the mock thing but not for `application.h`. So we may pick one also depending on the previous comments.

Reply:
To be honest, I have some doubts about the test framework CMake interfaces. The test framework ideally should behave as any other CMake dependency, i.e.: it should be used as a normal library, therefore test targets should be exposed and not hidden behind a ```test_program``` directive.

The reason is simple: if we expect to reuse the testing library across several projects, we risk having to re-expose the whole CMake APIs through our CMake calls (for example, how can I setup a preprocessor macro value on one of the tests? Will we implement a ```test_program_add_definitions```?). For this reason I think that having a ```test_program_include_directories``` is necessary, but it still is a workaround. 

Regarding the ```"application.h"``` situation, I think that if you consider the content of that folder as a separate application, than it should be built separately as a library with its own include header directories, to make sure it does not depend on any other of our sources. Otherwise it is just a normal test and it needs to obey to the rules of the rest of our source files.

Please, let me know what you think!

Reply:
This was noted in ROOT-Sim/testing-framework/issues/7

Reply:
Perfect, for the moment we go with this, if we want to refine the test framework interface (or use an off-the-shelf library for testing), we'll do that.